tardis/__init__.py

```diff
@@ -33,6 +33,126 @@
 logger.addHandler(console_handler)
 logging.getLogger("py.warnings").addHandler(console_handler)
 
+LOGGING_LEVELS = {
+    "NOTSET": logging.NOTSET,
+    "DEBUG": logging.DEBUG,
+    "INFO": logging.INFO,
+    "WARNING": logging.WARNING,
+    "ERROR": logging.ERROR,
+    "CRITICAL": logging.CRITICAL,
+}
+
+
+class FilterLog(object):
+    """
+    Filter Log Class for Filtering Logging Output
+    to a particular level
+
+    Parameters
+    ----------
+    log_level : logging object
+        allows to have a filter for the
+        particular log_level
+    """
+
+    def __init__(self, log_level):
+        self.log_level = log_level
+
+    def filter(self, log_record):
+        """
+        filter() allows to set the logging level for
+        all the record that are being parsed & hence remove those
+        which are not of the particular level
+
+        Parameters
+        ----------
+        log_record : logging.record
+            which the paricular record upon which the
+            filter will be applied
+
+        Returns
+        -------
+        boolean : True, if the current log_record has the
+            level that of the specified log_level
+            False, if the current log_record doesn't have the
+            same log_level as the specified one
+        """
+        return log_record.levelno == self.log_level
+
+
+# Setting up a list to store the Logging Filters set by logger.addFilter()
+list_of_filter = []
+
+
+def logging_state(log_state, tardis_config, specific):
+    """
+    Function to set the logging configuration for the simulation output
+    Called from within run_tardis()
+    Configured via functional arguments passed through run_tardis() - log_state & specific
+    Configured via YAML parameters under `debug` section - logging_level & specific_logging
+
+    Parameters
+    ----------
+    log_state: str
+        Allows to input the log level for the simulation
+        Uses Python logging framework to determine the messages that will be output
+    specific: boolean
+        Allows to set specific logging levels. Logs of the `log_state` level would be output.
+    """
+    if "debug" in tardis_config:
+        if specific or tardis_config["debug"]["specific"]:
+            specific = True
+        else:
+            specific = False
```

Comment:
This might be a little confusing from an end-user point of view because it will set `specific` to be true if it is specified in the function call or in the config.  You should have the priority for the selection be one or the other, rather than prioritize a value.  Additionally, this might be good reason to make `specific` a kwarg set to `None` by default, then check if a value is provided.  If a value is given, then use that value, if not use the value from config

```suggestion
         specific = tardis_config["debug"]["specific"] if specific is None else specific
```

In this code example, `specific` would be a keyword arg of the function with default value `None`.  If a value is given by the user, that value is used.  If not, then it uses the value from the config.

Reply:
Yes, absolutely. This change will make it easier for understanding the code & remove the redundancies that I have created via this implemented. Thank you very much for this suggestion 😄